crates/alerter/src/main.rs

```diff
@@ -25,11 +25,34 @@ struct Order {
     sell_token: H160,
     #[serde(with = "u256_decimal")]
     sell_amount: U256,
+    #[serde(with = "u256_decimal")]
+    fee_amount: U256,
     uid: OrderUid,
     status: OrderStatus,
     creation_date: DateTime<Utc>,
     partially_fillable: bool,
     is_liquidity_order: bool,
+    #[serde(flatten)]
+    class: OrderClass,
+}
+
+impl Order {
+    fn effective_sell_amount(&self) -> Option<U256> {
+        let amount = match &self.class {
+            OrderClass::Limit(limit) => {
+                // Use wrapping arithmetic. The orderbook should guarantee that
+                // the effective sell amount fits in a `U256`.
+                self.sell_amount
+                    .overflowing_add(self.fee_amount)
+                    .0
+                    .overflowing_sub(limit.surplus_fee?)
+                    .0
+            }
+            _ => self.sell_amount,
```

Comment:
Given your good and valid comment on #1012, should we start avoiding default implementations in match statements for enums? Will make it easier in the future when we add a new order class to revisit all call-sites

Reply:
Good point - updated.